Sources/SMTPKitten/Connection/SMTPClient.swift

```diff
@@ -0,0 +1,157 @@
+import NIOCore
+import NIOPosix
+import NIOExtras
+import NIOSSL
+
+public actor SMTPClient {
+    let channel: NIOAsyncChannel<SMTPReplyLine, ByteBuffer>
+    fileprivate let requests: AsyncStream<SMTPRequest>
+    fileprivate let requestWriter: AsyncStream<SMTPRequest>.Continuation
+    fileprivate var error: Error?
+    fileprivate var _handshake: SMTPHandshake?
+    internal var handshake: SMTPHandshake {
+        guard let _handshake else {
+            preconditionFailure("SMTPClient didn't set the SMTPHandshake after getting it")
+        }
+
+        return _handshake
+    }
+
+    fileprivate init(channel: NIOAsyncChannel<SMTPReplyLine, ByteBuffer>) {
+        self.channel = channel
+        (requests, requestWriter) = AsyncStream.makeStream(of: SMTPRequest.self, bufferingPolicy: .unbounded)
+    }
+
+    fileprivate func setHandshake(to handshake: SMTPHandshake) {
+        self._handshake = handshake
+    }
+
+    internal func send(_ request: ByteBuffer) async throws -> SMTPReply {
+        try await withCheckedThrowingContinuation { continuation in
+            let request = SMTPRequest(buffer: request, continuation: continuation)
+            requestWriter.yield(request)
+        }
+    }
+
+    internal func run() async throws {
+        do {
+            try await channel.executeThenClose { inbound, outbound in
+                var inboundIterator = inbound.makeAsyncIterator()
+
+                for await request in requests {
+                    do {
+                        if request.buffer.readableBytes > 0 {
+                            // The first "message" on a connection send by us is empty
+                            // Because we're expecting to read data here, not write
+                            try await outbound.write(request.buffer)
+                        }
+
+                        guard var lastLine = try await inboundIterator.next() else {
```

Comment:
Is it always the case except for the first incoming message, that you only get a response if you send a request?